### ISSUE TITLE
Instagram mock layout - Clarity needed on best practices

### ISSUE THREAD

**Issue by sree127** [Instagram/IGListKit#958]
Hi,

Below is an example of what I'm trying to do:

<img width="288" alt="screen shot 2017-10-06 at 8 40 31 am" src="https://user-images.githubusercontent.com/3157263/31261470-0df8e6e4-aa72-11e7-81ae-aef393af516f.png">

I could achieve something similar but I'm not sure whether its the right way to do that. Below is a rough structure of what I'm following:
 
Models:
```
class FeedEntry: ListDiffable {
    
    let id: String
    let date: Date
    let text: String
    let user: User
    
    init(id: String, date: Date, text: String, user: User) {
        self.id = id
        self.date = date
        self.text = text
        self.user = user
    }
    
    func diffIdentifier() -> NSObjectProtocol {
        return id as NSObjectProtocol
    }
    
    func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
        guard self !== object else { return true }
        guard let object = object as? FeedEntry else { return false }
        return id == object.id
    }
}
```

```
class User: NSObject, ListDiffable {
    
    let id: String
    let name: String
    
    init(id: String, name: String) {
        self.id = id
        self.name = name
    }
    
    // MARK: ListDiffable
    func diffIdentifier() -> NSObjectProtocol {
        return id as NSObjectProtocol
    }
    
    func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
        guard self !== object else { return true }
        guard let object = object as? User else { return false }
        return id == object.id && name == object.name
    }
}
```

```
/// Suggested Buddy List
    var suggestedBuddyList: [User] {
        var userArray = [User]()
        userArray.append(User(id: "1", name: "Sreejith"))
        userArray.append(User(id: "2", name: "Aasd"))
        userArray.append(User(id: "3", name: "CCwss"))
        userArray.append(User(id: "4", name: "VWWW"))
        return userArray
    }
    
    /// Load Feed
    var entries: [FeedEntry] {
        let user = User(id: "444", name: "Sreejith N")
        let entries = [
            FeedEntry(
                id: "11",
                date: Date(timeIntervalSinceNow: -1727283),
                text: "Ok I think I have this potato thing figured out. I'm using some of the leftover fuel from the landing thruster and basically lighting it on fire. The hydrogen and oxygen combine to make water. If I throttle the reaction I can let this run all day and generate enough water in the air to hydrate my potatos.\n\nThough, I'm basically igniting jet fuel in my living room.",
                user: user
            ),
            FeedEntry(
                id: "211",
                date: Date(timeIntervalSinceNow: -1382400),
                text: "I blew up.\n\nMy potato hydration system was working perfectly, but I forgot to account for excess oxygen from the reaction. I ended up with 30% pure oxygen in the HAB. Where I'm making mini explosions. Oh did I mention I live here?\n\nI survived but the HAB is basically gone, along with all my potatos. The cold air instantly froze the ones I have, so there's that at least.",
                user: user
            )
        ]
        return entries
    }
```

**I'm not quite sure whether the adapter datasource is configured properly:**

```
    func objects(for listAdapter: ListAdapter) -> [ListDiffable] {
        return [6, "1"] as! [ListDiffable]
    }
    
    func listAdapter(_ listAdapter: ListAdapter, sectionControllerFor object: Any) -> ListSectionController {
        
        if object is String {
            let feed = FeedSectionController()
            feed.entries = entries
            return feed
        } else {
            let horizontalSectionController = HorizontalSectionController()
            horizontalSectionController.buddy = suggestedBuddyList
            return horizontalSectionController
        }
    }
```

Basically what I'd like to achieve is load the data as:
```
   var data: [ListDiffable] = [ListDiffable]()

   data += suggestedBuddyList as [ListDiffable]
   data += entries as [ListDiffable]
```
and use the same `data` in `objects(for listAdapter: ListAdapter)`. 

The horizontal collection view should be at 0th index and will not repeat. Hence I dropped the idea of using a `stacklistcontroller`.  Can you tell me if this is the right approach? 

I'm attaching my project here for reference. 
[IGListTest.zip](https://github.com/Instagram/IGListKit/files/1361468/IGListTest.zip)

**Comment by jessesquires** [comment]
Thanks @sree127 !

StackOverflow is probably a more appropriate place for this question. 😄
